airbyte-integrations/connectors/source-faker/source_faker/purchase_generator.py

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+from multiprocessing import current_process
+from typing import Dict
+
+from airbyte_cdk.models import AirbyteRecordMessage, Type
+from mimesis import Datetime, Numeric
+
+from .airbyte_message_with_cached_json import AirbyteMessageWithCachedJSON
+from .utils import format_airbyte_time, now_millis
+
+
+class PurchaseGenerator:
+    def __init__(self, stream_name: str, seed: int) -> None:
+        self.stream_name = stream_name
+        self.seed = seed
+
+    def prepare(self):
+        """
+        Note: the instances of the mimesis generators need to be global.
+        Yes, they *should* be able to be instance variables on this class, which should only instantiated once-per-worker, but that's not quite the case:
+        * relying only on prepare as a pool initializer fails because we are calling the parent process's method, not the fork
+        * Calling prepare() as part of generate() (perhaps checking if self.person is set) and then `print(self, current_process()._identity, current_process().pid)` reveals multiple object IDs in the same process, resetting the internal random counters
+        """
+
+        seed_with_offset = self.seed
+        if self.seed is not None and len(current_process()._identity) > 0:
+            seed_with_offset = self.seed + current_process()._identity[0]
+
+        global dt
+        global numeric
```

Comment:
@sherifnada I was able to address about 1/2 of your comments.  For the rest, I added docs :D